src/components/ui/BaseSort.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+  <div class="lg:flex p-1 mt-4 sm:mx-3 mb-5 justify-between">
+    <div class="flex my-3 p-1 lg:h-10 sm:h-9">
+      <BaseButton
+        :class="{ 'bg-emerald-400 border-none': activeButton === 'title' }"
+        @click="activateButton('title')"
+        class="border-black border-2 rounded-lg font-display-pro leading-4 min-w-12 text-black px-2 text-xs"
+        >Title</BaseButton
+      >
+      <BaseButton
+        :class="{
+          'bg-emerald-400 border-none': activeButton === 'description',
+        }"
+        @click="activateButton('description')"
+        class="border-black border-2 rounded-lg font-display-pro max-w-24 leading-4 text-black ml-3 px-2 text-xs"
+        >Description</BaseButton
+      >
+      <BaseButton
+        class="border-black border-2 rounded-lg font-display-pro leading-4 max-w-16 ml-3 px-2 text-xs"
+        :class="{ 'bg-emerald-400 border-none': activeButton === 'priority' }"
+        @click="activateButton('priority')"
+        >Priority</BaseButton
+      >
+      <BaseButton
+        class="border-black border-2 rounded-lg font-display-pro leading-4 max-w-16 ml-3 px-2 text-xs"
+        :class="{ 'bg-emerald-400 border-none': activeButton === 'date' }"
+        @click="activateButton('date')"
+        >Date</BaseButton
+      >
+    </div>
+    <div class="sm:my-4 sm:justify-start">
+      <BaseButton class="ml-2">
+        <ArrowUpIcon
+          :class="{ 'bg-emerald-400 border-none': isAscActive }"
+          class="h-7 w-8 text-white px-2 py-1 rounded-lg bg-black"
+          @click="onSortedTodos('asc')"
+        ></ArrowUpIcon>
+      </BaseButton>
+
+      <BaseButton class="ml-2">
+        <ArrowDownIcon
+          :class="{ 'bg-emerald-400 border-none': isDescActive }"
+          class="h-7 w-8 text-white px-2 py-1 rounded-lg bg-black"
+          @click="onSortedTodos('desc')"
+        ></ArrowDownIcon>
+      </BaseButton>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref } from "vue";
+import BaseButton from "./BaseButton.vue";
+import { ArrowUpIcon } from "@heroicons/vue/24/solid";
+import { ArrowDownIcon } from "@heroicons/vue/24/solid";
+
+const activeButton = ref("");
+const isAscActive = ref(false);
```

Comment:
I would create an activeDirection variable, which can be empty string, asc or desc. In that case you dont need to have two variables for direction